cellxgene_resources/curation_visium.ipynb

```diff
@@ -0,0 +1,227 @@
+{
+ "cells": [
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import anndata as ad\n",
+    "import numpy as np\n",
+    "import pandas as pd\n",
+    "import scanpy as sc\n",
+    "from PIL import Image\n",
+    "from scipy import sparse"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Specify the folder that includes these files...\n",
+    "- raw_feature_bc_matrix.h5\n",
+    "- spatial/\n",
+    "  - scalefactors_json.json\n",
+    "  - tissue_hires_image.png\n",
+    "  - tissue_lowres_image.png*\n",
+    "  - tissue_positions_list.csv\n",
+    "\n",
+    "*the lowres image will be deleted but is required for `read_visium()`"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "folder = 'WS_PLA_S9101764'\n",
+    "\n",
+    "#REQUIRED to include empty spots, so specify raw_feature_bc_matrix.h5\n",
+    "adata = sc.read_visium(folder, count_file='raw_feature_bc_matrix.h5')"
```

Comment:
The code block below to fill in empty 0s initially failed until I ran `adata.var_names_make_unique()`. I think further comment to clarify to run this method if the size is < 4992 and there is a warning that variable names are not unique.

Reply:
Good catch. Fixed by setting gene_ids as the index to start.

Reply:
Looks good, looks like this now works correctly when obs is < 4992